File: leetcode/125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        
        int i =0, j = s.size()-1;

        while( i < j)
        {
            while( i < s.size() &&  !isalnum( (int)s.at(i)) )
            {
                i++;
            }
            
            while(j > -1 &&  !isalnum( (int)s.at(j)) )
                j--;

            if( i < s.size() && j > -1)
            {
                int a = tolower( s.at(i));
                int b = tolower( s.at(j));

                if( a == b)
                {
                    i++;
                    j--;
                }
                else
                {
                    return false;
                }
            }
        }

        return true;

    }
};

File: leetcode/125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        
        i =0
        j = len(s)-1
        s = s.lower()
        while i<j:
            while i<len(s) and not s[i].isalnum():
                i+=1
            while j>=0 and not s[j].isalnum():
                j-=1
            
            if i>=len(s) and j <0 : 
                return True

            if  s[i] == s[j] :
                i+=1
                j-=1
            else:
                return False
        
        return True